compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -1440,7 +1440,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         if !ty.is_ty_var() {
             ty
         } else {
-            if !self.is_tainted_by_errors() {
+            if let None = self.tainted_by_errors() {
```

Comment:
can you change this `if` to return `ErrorGuaranteed` and use that in the `ty_error` right after that `if`?